src/Weather/BackgroundDownloadProgress.cpp

```
@@ -0,0 +1,195 @@
+void
+BackgroundDownloadProgress::Begin(const char *text) noexcept
+{
+  const unsigned prev = active_sessions.fetch_add(1);
+  {
+    const std::lock_guard lock{text_mutex};
+    pending_text = text;
+    if (prev == 0)
+      pending_show.store(true, std::memory_order_relaxed);
+    else
+      text_pending = true;
+  }
```

Cppcheck (2.21.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)